src/Chapter3.hs

```diff
@@ -907,9 +1010,23 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+
+newtype Gold = MkGold { unGold :: Int }
+
 class Append a where
     append :: a -> a -> a
 
+instance Append Gold where
+  append (MkGold a) (MkGold b) = MkGold $ a + b
+
+instance Append [a] where
+  append = (++)
+
+instance Append a => Append (Maybe a) where
+  append (Just a) (Just b) = Just $ a `append` b
+  append (Just a) Nothing = Just a
+  append (Nothing) (Just a) = Just a
+  append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂

Reply:
Good call! Thanks for pointing that out! I'll make that change for sure